### ISSUE TITLE
[local_docker] Distributed jobs do not work with default config

### ISSUE THREAD

**Issue by aivanou** [pytorch/torchx#286]
Distributed jobs with `local_docker` scheduler do not work. Each docker container gets allocated with default network `bridge`, that makes them not available to communicate between each other.
Also, the name and hostname is not assigned to each docker container. This means that `c10d` rendezvous cannot be used since we cannot provide a know master address.

Proposal.

The proposal is to add a new `network` option to the local scheduler `runopts`. If set, this option will be passed to docker as `--net` parameter . This will make sure that all containers launched by the `local_docker` will be launched in the same network and all ports will be available between them.

Add `--name` and `--hostname` to the docker command with value : `${APP_NAME}-${ROLE_NAME}-${REPLICA_ID}`.
This make sure that each docker container will have a unique hostname that can be referred in the command line, e.g.

```

torchx run -s local_docker\
 ./torchx/examples/apps/lightning_classy_vision/component.py:trainer_dist \
--nnodes 2 --nproc_per_node 2 \
--rdzv_backend c10d \
--rdzv_endpoint cv-trainer-worker-0:29500

```

**Comment by d4l3k** [comment]
With the default bridge network distributed jobs should be able to communicate with each other but they'll have to use the bridge IPs instead of localhost.

https://docs.docker.com/network/bridge/

Launching an etcd node on the host and then doing using the hostname of the host should allow the workers to find each other and train 

Need to do some more testing here
